### PR TITLE
Bugfix and refactor

### PR DESCRIPTION
**Pull request by zxdavb**
This PR removed the `print()` and uses `logger()` instead.  It also has some de-linting.

It cleans up `ZoneBase` inheritance.

Try this:
```python
from evohomeclient2 import EvohomeClient
c = EvohomeClient(username, password, refresh_token=rt)
c.zone_schedules_backup("schedule.json")
```
you get:
```
INFO:evohomeclient2.controlsystem:Backing up schedules from ControlSystem: 3432522 (My Home)...
INFO:evohomeclient2.controlsystem:Retrieving Zone schedule: 3432521 - Main Room
INFO:evohomeclient2.controlsystem:Retrieving Zone schedule: 3432576 - Front Room
INFO:evohomeclient2.controlsystem:Retrieving Zone schedule: 3432577 - Kitchen
INFO:evohomeclient2.controlsystem:Retrieving Zone schedule: 3432578 - Bedroom
INFO:evohomeclient2.controlsystem:Retrieving Zone schedule: 3432579 - Beans Room
INFO:evohomeclient2.controlsystem:Retrieving Zone schedule: 3432580 - Noos Room
INFO:evohomeclient2.controlsystem:Retrieving Zone schedule: 3449703 - Bathroom
INFO:evohomeclient2.controlsystem:Retrieving Zone schedule: 3449740 - Test
INFO:evohomeclient2.controlsystem:Retrieving Zone schedule: 3450733 - PIV Output
INFO:evohomeclient2.controlsystem:Writing to backup file: schedule.json...
INFO:evohomeclient2.controlsystem:Backup completed.
```

### CODE REVIEW

File: evohomeclient2/controlsystem.py [hunk 1]
```diff
@@ -112,38 +115,46 @@ def temperatures(self):
...
             file_output.write(schedule_db)
 
-        print("Backed up zone schedule to: %s" % filename)
+        _LOGGER.info("Backup completed.")
 
     def zone_schedules_restore(self, filename):
-        """Restores all zones on control system from the given file"""
-        print("Restoring zone schedules from: %s" % filename)
+        """Restore all zones on control system from the given file."""
+        _LOGGER.setLevel(logging.INFO)
```
Comment (watchforstock) [inline evohomeclient2/controlsystem.py]: Just wanted to check whether we want to be forcing the logging level here? Will this not override the users preferences set elsewhere?

File: evohomeclient2/controlsystem.py [hunk 2]
```diff
@@ -112,38 +115,46 @@ def temperatures(self):
...
+        _LOGGER.warning("Restoring schedules to ControlSystem %s (%s)...",
```
Comment (watchforstock) [inline evohomeclient2/controlsystem.py]: This feels like it should still be at info as it's not warning about a failure somewhere

Comment (zxdavb) [reply evohomeclient2/controlsystem.py]: I put it as a warning becuase it was making a change, but after reading (https://docs.python.org/3/howto/logging.html), I think you're right - should be `logging.INFO`.

File: evohomeclient2/controlsystem.py (hunk 1 above)
Comment (zxdavb) [reply evohomeclient2/controlsystem.py]: Yeah, these (there are two) shouldn't have been left in.  I will fix it.

**Comment by zxdavb** [comment]
@watchforstock I got into some rebase disaster (again) - are yuo happt to accept this PR without a squash.

FWIW, I have no plans for any more PRs!

**Pull request merged**